### DISCUSSION TITLE
20230918-rallit-standalone/

### DISCUSSION THREAD

**Discussion by giscus** [inflearn/tech-blog-comment#43] [Comments]
# 20230918-rallit-standalone/

https://tech.inflab.com/20230918-rallit-standalone/

**Comment by Chanki-Min** [comment]
저도 비슷한 상황 (dd-trace, 및 기타 preload 필요)에서 standalone을 적용했었는데, 이렇게 보니 반갑네요.

next 13.1.x 버전까지는 dd-trace가 잘 작동했던거로 기억하고 있는데, preload된 라이브러리들은 vercel nft가 사용하는 라이브러리 인식하지 않고 standalone 패키지에서 제외되는 문제가 있었고, dockerfile에서 global install 후 node process 환경변수에 global module path를 지정해서 해결했었습니다.

next config에서 nft에 포함시킬 라이브러리 지정이 쉽게 되면 좋겠는데 아직 개발중인 단계라 직접 만져줘야 하는 것들이 많은 것 같아요.

**Reply by gohsue-inflab** [reply]
인사가 늦었습니다! 말씀대로 standalone 패키지에서 제외되는 라이브러리나, 사전에 preload가 필요한 라이브러리는 직접 넣어줘야하는 번거로움이 있었습니다.
dd-trace 같은 경우에 `require('dd-trace/init')` 으로 경로를 수정해주면 next 13버전에서도 Datadog APM을 사용할 수 있습니다. 참고가 되었으면 좋겠습니다! 😄 문서는 조만간 반영하도록 하겠습니다!

**Comment by kindunq** [comment]
안녕하세요 저희도 비슷한 사유로 standalone방식을 적용하던 차에 블로그를 방문했습니다

덕분에 잘 적용했습니다 
차이점은 저희는 npm으로 build하고있고

build:dev : env-cmd -f ./env/.env.development next build

이렇게 빌드를 하면 env파일 이동없이도 환경변수를 인식하더라고요

진짜 알수록 무지해집니다.

**Reply by gohsue-inflab** [reply]
안녕하세요! 인사가 늦었습니다.
포스팅이 도움이 되었다니 영광입니다 😄

env-cmd를 이용한 방식을 최초에 시도했으나, 동작하지 않아 이와 같은 전처리 과정을 추가했는데요.
혹시나 하는 마음으로 다시 실행해봤으나 env가 누락되네요 😂 npm에서는 제대로 동작하다니 다행입니다.

코멘트 감사합니다! npm 사용하시는 분들은 참고가 되었으면 좋겠습니다.
